src/rgw/rgw_rest_user.cc

```diff
@@ -183,6 +183,9 @@ void RGWOp_User_Create::execute()
   }
 
   if (max_buckets != default_max_buckets)
```

Comment:
@cbodley note that if statement doesn't open a scope, so this is completely broken now

Reply:
it makes the following statement depend on the condition;  the following statement is I think "if (max_buckets < 0) { max_buckets =-1; }"  then op_state.set_max_buckets(max_buckets); is executed unconditionally
no?\3

Reply:
@mattbenjamin yup, this needs to be fixed.

Reply:
just came across this. because the GCC warning. see #32884